unit_test/sparse/Test_Sparse_block_gauss_seidel.hpp

```diff
@@ -143,37 +139,33 @@ int run_block_gauss_seidel_1(
   return 0;
 }
 
-template<typename scalar_view_t>
-scalar_view_t create_x_vector(size_t nv, double max_value = 10.0){
-  scalar_view_t kok_x ("X", nv);
-
-
-  typename scalar_view_t::HostMirror h_x =  Kokkos::create_mirror_view (kok_x);
-
-
-  for (size_t i = 0; i < nv; ++i){
-    typename scalar_view_t::value_type r =
-        static_cast <typename scalar_view_t::value_type> (rand()) /
-        static_cast <typename scalar_view_t::value_type> (RAND_MAX / max_value);
-    h_x(i) = r;
-    //h_x(i) = 1;
+template<typename vec_t>
+vec_t create_x_vector(vec_t& kok_x, double max_value = 10.0) {
+  typedef typename vec_t::value_type scalar_t;
+  auto h_x = Kokkos::create_mirror_view (kok_x);
+  for (size_t j = 0; j < h_x.extent(1); ++j){
+    for (size_t i = 0; i < h_x.extent(0); ++i){
+      scalar_t r =
+          static_cast <scalar_t> (rand()) /
+          static_cast <scalar_t> (RAND_MAX / max_value);
+      h_x.access(i, j) = r;
```

Comment:
Again, it's weird that you call `access` -- how come?

Reply:
@mhoemmen Originally I wanted the implementation to support rank-1 and rank-2 inputs, but that's not actually necessary since I now wrap the user's view in a rank-2 no matter what. I'll fix this.